lib/notifications.dart

```diff
@@ -71,8 +86,40 @@ class NotificationService {
     token.value = value;
   }
 
-  static void _onRemoteMessage(FirebaseRemoteMessage message) {
+  static void _onForegroundMessage(FirebaseRemoteMessage message) {
     assert(debugLog("notif message: ${message.data}"));
+    _onRemoteMessage(message);
+  }
+
+  static Future<void> _onBackgroundMessage(FirebaseRemoteMessage message) async {
+    // This callback will run in a separate isolate from the rest of the app.
+    // See docs:
+    //   https://firebase.flutter.dev/docs/messaging/usage/#background-messages
+    _initBackgroundIsolate();
+
+    assert(debugLog("notif message in background: ${message.data}"));
+    _onRemoteMessage(message);
+  }
+
+  static void _initBackgroundIsolate() {
+    bool isolateIsLive = true;
+    assert(() {
+      isolateIsLive = debugBackgroundIsolateIsLive;
+      return true;
+    }());
+    if (!isolateIsLive) {
+      return;
+    }
+
+    assert(() {
+      debugLogEnabled = true;
+      return true;
+    }());
```

Comment:
What's the reason for setting `debugLogEnabled` to true here?

Reply:
Same as in `main` (the one in `lib/main.dart`) — it causes the `debugLog` calls to be active when running a debug build of the app, so that you get them in `flutter run` output.

Or perhaps the question is: why does this code need to do that, when the whole rest of the app is covered by `main` doing so? That's because it's in a separate isolate, as mentioned in `_onBackgroundMessage`; same reason it needs the `LiveZulipBinding.ensureInitialized` and `NotificationDisplayManager._init` calls.

I'll add a comment pointing out the comparison to `main`.